app/controllers/repos_controller.rb

```diff
@@ -41,13 +41,26 @@ def create
     @repo   = Repo.search_by(params[:repo][:name], params[:repo][:user_name]).first unless params_blank?
     @repo ||= Repo.new(repo_params)
     if @repo.save
-      @repo_sub = current_user.repo_subscriptions.create(repo: @repo)
-      flash[:notice] = "Added #{@repo.to_param} for triaging"
-      RepoSubscription.background_send_triage_email(@repo_sub.id)
-      redirect_to @repo
+      @repo_sub = current_user.repo_subscriptions.find_or_initialize_by(repo: @repo)
+      
+      if @repo_sub.new_record?
```

Comment:
This part is still relevant since this is the ones that handles the flashing of the right flash message after clicking the repo subscription